cirq/ops/eigen_gate.py

```diff
@@ -349,8 +352,10 @@ def _is_parameterized_(self) -> bool:
     def _parameter_names_(self) -> AbstractSet[str]:
         return protocols.parameter_names(self._exponent)
 
-    def _resolve_parameters_(self: TSelf, param_resolver, recursive: bool) -> 'EigenGate':
-        return self._with_exponent(exponent=param_resolver.value_of(self._exponent, recursive))
+    def _resolve_parameters_(
+        self: TSelf, resolver: 'cirq.ParamResolver', recursive: bool
+    ) -> 'EigenGate':
```

Comment:
Should this be `TSelf` instead of `EigenGate`, to match `self`?

Reply:
I tried that, but it doesn't work because `self._with_exponent` does not return `TSelf`. If you change that also, there are problems with eigengate subclasses like `_PauliX` since those can return a different type when the exponent is changed (e.g. a generic `XPowGate` in stead of `_PauliX`). I think we'll have to live with the less specific types here. I did remove the `TSelf` annotation here since it wasn't doing anything.

Reply:
SGTM.